packages/helix-shared-indexer/src/index-resource.js

```diff
@@ -177,11 +177,19 @@ export function indexResource(path, response, config, log) {
 
   /* Walk through all index properties */
   config.properties.forEach((property) => {
-    const { select, ...p } = property;
+    const { select, selectFirst, ...p } = property;
     const expression = p.value || p.values;
-    // create an array of elements
+
     try {
-      const elements = select !== 'none' ? selectAll(select, content) : [];
+      let elements = [];
+      if (selectFirst) {
+        const element = selectOne(selectFirst, content);
+        if (element) {
+          elements = [element];
+        }
+      } else if (select !== 'none') {
+        elements = selectAll(select, content);
+      }
```

Comment:
new behaviour for `selectFirst`: select first, only turn into array if result is not `undefined`